<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="hoodscanR">
<title>A quick start guide to the hoodscanR package • hoodscanR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="A quick start guide to the hoodscanR package">
<meta property="og:description" content="hoodscanR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">hoodscanR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Quick_start.html">A quick start guide to the hoodscanR package</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/DavisLaboratory/hoodscanR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>A quick start guide to the hoodscanR package</h1>
                        <h4 data-toc-skip class="author">Ning Liu,
Melissa Davis</h4>
            
            <h4 data-toc-skip class="date">2023-09-15</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/DavisLaboratory/hoodscanR/blob/HEAD/vignettes/Quick_start.Rmd" class="external-link"><code>vignettes/Quick_start.Rmd</code></a></small>
      <div class="d-none name"><code>Quick_start.Rmd</code></div>
    </div>

    
    
<style type="text/css">
  body{
  font-size: 14pt;
}
</style>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>hoodscanR is an user-friendly R package providing functions to assist
cellular neighborhood analysis of any spatial transcriptomics data with
single-cell resolution.</p>
<p>All functions in the package are built based on the SpatialExperiment
infrastructure, allowing integration into various spatial
transcriptomics-related packages from Bioconductor. The package can
result in cell-level neighborhood annotation output, along with funtions
to perform neighborhood colocalization analysis and neighborhood-based
cell clustering.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="st">"hoodscanR"</span><span class="op">)</span></span></code></pre></div>
<p>The development version of <code>hoodscanR</code> can be installed
from GitHub:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"DavisLaboratory/hoodscanR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="quick-start">Quick start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/DavisLaboratory/hoodscanR" class="external-link">hoodscanR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drighelli/SpatialExperiment" class="external-link">SpatialExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thomasp85/scico" class="external-link">scico</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-exploration">Data exploration<a class="anchor" aria-label="anchor" href="#data-exploration"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"spe_test"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readHoodData.html">readHoodData</a></span><span class="op">(</span><span class="va">spe</span>, anno_col <span class="op">=</span> <span class="st">"celltypes"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SpatialExperiment </span></span>
<span><span class="co">## dim: 50 2661 </span></span>
<span><span class="co">## metadata(1): dummy</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(50): MERTK MRC1 ... SAA2 FZD4</span></span>
<span><span class="co">## rowData names(0):</span></span>
<span><span class="co">## colnames(2661): Lung9_Rep1_5_5 Lung9_Rep1_5_6 ... Lung9_Rep1_5_4047</span></span>
<span><span class="co">##   Lung9_Rep1_5_4052</span></span>
<span><span class="co">## colData names(9): orig.ident nCount_RNA ... cell_annotation sample_id</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## spatialCoords names(2) : x y</span></span>
<span><span class="co">## imgData names(0):</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">colData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 2661 rows and 9 columns</span></span>
<span><span class="co">##                   orig.ident nCount_RNA nFeature_RNA       fov      Area</span></span>
<span><span class="co">##                     &lt;factor&gt;  &lt;numeric&gt;    &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span></span>
<span><span class="co">## Lung9_Rep1_5_5         Lung9        182          104         5      4377</span></span>
<span><span class="co">## Lung9_Rep1_5_6         Lung9        447          214         5      4678</span></span>
<span><span class="co">## Lung9_Rep1_5_7         Lung9        234          148         5      2236</span></span>
<span><span class="co">## Lung9_Rep1_5_8         Lung9        118           69         5      4781</span></span>
<span><span class="co">## Lung9_Rep1_5_14        Lung9        424          235         5      4385</span></span>
<span><span class="co">## ...                      ...        ...          ...       ...       ...</span></span>
<span><span class="co">## Lung9_Rep1_5_4040      Lung9         84           65         5      1720</span></span>
<span><span class="co">## Lung9_Rep1_5_4041      Lung9        153          105         5      3418</span></span>
<span><span class="co">## Lung9_Rep1_5_4045      Lung9         48           42         5      1735</span></span>
<span><span class="co">## Lung9_Rep1_5_4047      Lung9         50           41         5      2101</span></span>
<span><span class="co">## Lung9_Rep1_5_4052      Lung9         48           42         5       977</span></span>
<span><span class="co">##                   AspectRatio       slide cell_annotation   sample_id</span></span>
<span><span class="co">##                     &lt;numeric&gt; &lt;character&gt;     &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">## Lung9_Rep1_5_5           1.15  Lung9_Rep1     Tumor.cells    sample01</span></span>
<span><span class="co">## Lung9_Rep1_5_6           0.98  Lung9_Rep1     Tumor.cells    sample01</span></span>
<span><span class="co">## Lung9_Rep1_5_7           1.29  Lung9_Rep1     Tumor.cells    sample01</span></span>
<span><span class="co">## Lung9_Rep1_5_8           1.74  Lung9_Rep1         Stromal    sample01</span></span>
<span><span class="co">## Lung9_Rep1_5_14          1.11  Lung9_Rep1     Tumor.cells    sample01</span></span>
<span><span class="co">## ...                       ...         ...             ...         ...</span></span>
<span><span class="co">## Lung9_Rep1_5_4040        0.87  Lung9_Rep1 Epithelial.cell    sample01</span></span>
<span><span class="co">## Lung9_Rep1_5_4041        2.89  Lung9_Rep1  Dividing.cells    sample01</span></span>
<span><span class="co">## Lung9_Rep1_5_4045        1.45  Lung9_Rep1     Tumor.cells    sample01</span></span>
<span><span class="co">## Lung9_Rep1_5_4047        2.97  Lung9_Rep1 Epithelial.cell    sample01</span></span>
<span><span class="co">## Lung9_Rep1_5_4052        2.22  Lung9_Rep1 Epithelial.cell    sample01</span></span></code></pre>
<p>We can have a look at the tissue and cell positions by using the
function <code>plotTissue</code>.</p>
<p>The test data is relatively sparse with low-level cell type
annotations.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">col.pal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red3"</span>, <span class="st">"royalblue"</span>, <span class="st">"gold"</span>, <span class="st">"cyan2"</span>, <span class="st">"purple3"</span>, <span class="st">"darkgreen"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plotTissue.html">plotTissue</a></span><span class="op">(</span><span class="va">spe</span>, color <span class="op">=</span> <span class="va">cell_annotation</span>, size <span class="op">=</span> <span class="fl">1.5</span>, alpha <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="va">col.pal</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-8-1.png" width="560"></p>
</div>
<div class="section level2">
<h2 id="neighborhoods-scanning">Neighborhoods scanning<a class="anchor" aria-label="anchor" href="#neighborhoods-scanning"></a>
</h2>
<p>In order to perform neighborhood scanning, we need to firstly
identify k (in this example, k = 100) nearest cells for each cells.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fnc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findNearCells.html">findNearCells</a></span><span class="op">(</span><span class="va">spe</span>, k <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span></code></pre></div>
<p>The output of <code>findNearCells</code> function includes two
matrix, an annotation matrix and a distance matrix.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">fnc</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $cells</span></span>
<span><span class="co">##                 nearest_cell_1 nearest_cell_2  nearest_cell_3 nearest_cell_4</span></span>
<span><span class="co">## Lung9_Rep1_5_5     Tumor.cells    Tumor.cells     Tumor.cells    Tumor.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_6     Tumor.cells    Tumor.cells     Tumor.cells    Tumor.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_7     Tumor.cells    Tumor.cells     Tumor.cells Dividing.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_8         Stromal Dividing.cells Epithelial.cell        Stromal</span></span>
<span><span class="co">## Lung9_Rep1_5_14    Tumor.cells    Tumor.cells  Dividing.cells Dividing.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_15    Tumor.cells    Tumor.cells     Tumor.cells Dividing.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_18        Stromal        Stromal Epithelial.cell        Stromal</span></span>
<span><span class="co">## Lung9_Rep1_5_21    Tumor.cells    Tumor.cells     Tumor.cells    Tumor.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_22 Dividing.cells    Tumor.cells     Tumor.cells    Tumor.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_25        Stromal        Stromal         Stromal        Stromal</span></span>
<span><span class="co">##                 nearest_cell_5</span></span>
<span><span class="co">## Lung9_Rep1_5_5     Tumor.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_6     Tumor.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_7     Tumor.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_8  Dividing.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_14    Tumor.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_15    Tumor.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_18    Tumor.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_21 Dividing.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_22    Tumor.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_25        Stromal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $distance</span></span>
<span><span class="co">##                 nearest_cell_1 nearest_cell_2 nearest_cell_3 nearest_cell_4</span></span>
<span><span class="co">## Lung9_Rep1_5_5        89.89994      124.91997      145.38225      157.13688</span></span>
<span><span class="co">## Lung9_Rep1_5_6        43.41659       77.52419       84.97058       89.02247</span></span>
<span><span class="co">## Lung9_Rep1_5_7        43.41659       48.04165       97.00000      109.98636</span></span>
<span><span class="co">## Lung9_Rep1_5_8        64.40497      166.92813      202.35612      206.20621</span></span>
<span><span class="co">## Lung9_Rep1_5_14       71.80529       77.31753      115.20851      125.29964</span></span>
<span><span class="co">## Lung9_Rep1_5_15       56.08921      105.75916      108.66922      112.78741</span></span>
<span><span class="co">## Lung9_Rep1_5_18       99.80982      181.68654      186.81542      189.44656</span></span>
<span><span class="co">## Lung9_Rep1_5_21       46.09772       81.56592       86.57944       99.32271</span></span>
<span><span class="co">## Lung9_Rep1_5_22       29.73214       43.01163       49.64877       67.06713</span></span>
<span><span class="co">## Lung9_Rep1_5_25       87.70975      158.91193      181.03315      205.38014</span></span>
<span><span class="co">##                 nearest_cell_5</span></span>
<span><span class="co">## Lung9_Rep1_5_5        158.6852</span></span>
<span><span class="co">## Lung9_Rep1_5_6        101.4347</span></span>
<span><span class="co">## Lung9_Rep1_5_7        120.9339</span></span>
<span><span class="co">## Lung9_Rep1_5_8        228.7204</span></span>
<span><span class="co">## Lung9_Rep1_5_14       129.0349</span></span>
<span><span class="co">## Lung9_Rep1_5_15       143.0035</span></span>
<span><span class="co">## Lung9_Rep1_5_18       220.7374</span></span>
<span><span class="co">## Lung9_Rep1_5_21       108.4066</span></span>
<span><span class="co">## Lung9_Rep1_5_22        69.8570</span></span>
<span><span class="co">## Lung9_Rep1_5_25       276.8267</span></span></code></pre>
<p>We can then perform neighborhood analysis using the function
<code>scanHoods</code>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scanHoods.html">scanHoods</a></span><span class="op">(</span><span class="va">fnc</span><span class="op">$</span><span class="va">distance</span><span class="op">)</span></span></code></pre></div>
<p>The resulting matrix is the probability of each cell associating with
their 100 nearest cells.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pm</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##                 nearest_cell_1 nearest_cell_2 nearest_cell_3 nearest_cell_4</span></span>
<span><span class="co">## Lung9_Rep1_5_5      0.18304483     0.13150067     0.10311690     0.08819389</span></span>
<span><span class="co">## Lung9_Rep1_5_6      0.11420320     0.09526285     0.09032795     0.08757140</span></span>
<span><span class="co">## Lung9_Rep1_5_7      0.13475921     0.13227645     0.09680770     0.08601822</span></span>
<span><span class="co">## Lung9_Rep1_5_8      0.44211502     0.15585801     0.08768955     0.08183066</span></span>
<span><span class="co">## Lung9_Rep1_5_14     0.09572989     0.09233235     0.06700040     0.06022003</span></span>
<span><span class="co">## Lung9_Rep1_5_15     0.17384013     0.12208690     0.11878336     0.11411524</span></span>
<span><span class="co">## Lung9_Rep1_5_18     0.41139452     0.14935678     0.13744881     0.13159514</span></span>
<span><span class="co">## Lung9_Rep1_5_21     0.09624026     0.07886930     0.07599996     0.06848324</span></span>
<span><span class="co">## Lung9_Rep1_5_22     0.08024218     0.07690592     0.07485455     0.06845482</span></span>
<span><span class="co">## Lung9_Rep1_5_25     0.44586128     0.20603613     0.14803730     0.09789334</span></span>
<span><span class="co">##                 nearest_cell_5</span></span>
<span><span class="co">## Lung9_Rep1_5_5      0.08631823</span></span>
<span><span class="co">## Lung9_Rep1_5_6      0.07892756</span></span>
<span><span class="co">## Lung9_Rep1_5_7      0.07697076</span></span>
<span><span class="co">## Lung9_Rep1_5_8      0.05320675</span></span>
<span><span class="co">## Lung9_Rep1_5_14     0.05775686</span></span>
<span><span class="co">## Lung9_Rep1_5_15     0.08124211</span></span>
<span><span class="co">## Lung9_Rep1_5_18     0.07485027</span></span>
<span><span class="co">## Lung9_Rep1_5_21     0.06303149</span></span>
<span><span class="co">## Lung9_Rep1_5_22     0.06731485</span></span>
<span><span class="co">## Lung9_Rep1_5_25     0.02152751</span></span></code></pre>
<p>We can then merge the probabilities by the cell types of the 100
nearest cells.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hoods</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mergeByGroup.html">mergeByGroup</a></span><span class="op">(</span><span class="va">pm</span>, <span class="va">fnc</span><span class="op">$</span><span class="va">cells</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hoods</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##                 Dividing.cells Endothelial.cell Epithelial.cell  Immune.cell</span></span>
<span><span class="co">## Lung9_Rep1_5_5     0.006003838     3.564371e-04    1.063262e-05 3.133032e-02</span></span>
<span><span class="co">## Lung9_Rep1_5_6     0.206692970     5.075441e-06    1.381476e-02 4.447312e-07</span></span>
<span><span class="co">## Lung9_Rep1_5_7     0.263126690     9.512496e-07    2.015171e-02 6.336489e-08</span></span>
<span><span class="co">## Lung9_Rep1_5_8     0.216159266     1.481687e-06    1.337012e-01 3.953512e-09</span></span>
<span><span class="co">## Lung9_Rep1_5_14    0.174747009     7.946121e-03    1.972819e-03 9.591948e-06</span></span>
<span><span class="co">## Lung9_Rep1_5_15    0.117429056     4.244536e-04    6.713550e-04 2.374688e-03</span></span>
<span><span class="co">## Lung9_Rep1_5_18    0.024023243     4.349856e-06    1.376427e-01 1.394815e-07</span></span>
<span><span class="co">## Lung9_Rep1_5_21    0.206271880     4.045150e-04    5.811456e-02 0.000000e+00</span></span>
<span><span class="co">## Lung9_Rep1_5_22    0.218030798     5.962029e-03    2.615948e-02 2.456481e-07</span></span>
<span><span class="co">## Lung9_Rep1_5_25    0.014854723     4.151471e-07    9.489670e-04 0.000000e+00</span></span>
<span><span class="co">##                      Stromal Tumor.cells</span></span>
<span><span class="co">## Lung9_Rep1_5_5  0.000000e+00 0.962298768</span></span>
<span><span class="co">## Lung9_Rep1_5_6  0.000000e+00 0.779486754</span></span>
<span><span class="co">## Lung9_Rep1_5_7  0.000000e+00 0.716720587</span></span>
<span><span class="co">## Lung9_Rep1_5_8  5.802381e-01 0.069899943</span></span>
<span><span class="co">## Lung9_Rep1_5_14 2.589378e-05 0.815298565</span></span>
<span><span class="co">## Lung9_Rep1_5_15 0.000000e+00 0.879100448</span></span>
<span><span class="co">## Lung9_Rep1_5_18 7.565452e-01 0.081784355</span></span>
<span><span class="co">## Lung9_Rep1_5_21 1.453902e-01 0.589818860</span></span>
<span><span class="co">## Lung9_Rep1_5_22 2.321371e-03 0.747526073</span></span>
<span><span class="co">## Lung9_Rep1_5_25 9.774651e-01 0.006730819</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="neighborhoods-analysis">Neighborhoods analysis<a class="anchor" aria-label="anchor" href="#neighborhoods-analysis"></a>
</h2>
<p>We plot randomly plot 10 cells to see the output of neighborhood
scanning using <code>plotHoodMat</code>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotHoodMat.html">plotHoodMat</a></span><span class="op">(</span><span class="va">hoods</span>, n <span class="op">=</span> <span class="fl">10</span>, hm_height <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-15-1.png" width="400"></p>
<p>Or to check the cells-of-interest with the parameter
<code>targetCells</code> within the function</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotHoodMat.html">plotHoodMat</a></span><span class="op">(</span><span class="va">hoods</span>, targetCells <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Lung9_Rep1_5_1975"</span>, <span class="st">"Lung9_Rep1_5_2712"</span><span class="op">)</span>, hm_height <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-16-1.png" width="400"></p>
<p>We can then merge the neighborhood results with the SpatialExperiment
object using <code>mergeHoodSpe</code> so that we can conduct more
neighborhood-related analysis.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mergeHoodSpe.html">mergeHoodSpe</a></span><span class="op">(</span><span class="va">spe</span>, <span class="va">hoods</span><span class="op">)</span></span></code></pre></div>
<p>To summarise our neighborhood results, we can use
<code>calcMetrics</code> to calculate entropy and perplexity of the
probability matrix so that we can have a summarisatino of the
neighborhood distribution across the tissue slide, i.e. where
neighborhood is more distinct and where is more mixed.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calcMetrics.html">calcMetrics</a></span><span class="op">(</span><span class="va">spe</span>, pm_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">hoods</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>We then again use <code>plotTissue</code> to plot out the entropy or
perplexity.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotTissue.html">plotTissue</a></span><span class="op">(</span><span class="va">spe</span>, size <span class="op">=</span> <span class="fl">1.5</span>, color <span class="op">=</span> <span class="va">entropy</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/scico/man/ggplot2-scales.html" class="external-link">scale_color_scico</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"tokyo"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-19-1.png" width="560"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotTissue.html">plotTissue</a></span><span class="op">(</span><span class="va">spe</span>, size <span class="op">=</span> <span class="fl">1.5</span>, color <span class="op">=</span> <span class="va">perplexity</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/scico/man/ggplot2-scales.html" class="external-link">scale_color_scico</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"tokyo"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-20-1.png" width="560"></p>
<p>We can perform neighborhood colocalization analysis using
<code>plotColocal</code>. Here we can see in the test data, endothelial
cells and stromal cells are more likely to colocalize, epithelial cells
and dividing cells are more likely to colocalize.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotColocal.html">plotColocal</a></span><span class="op">(</span><span class="va">spe</span>, pm_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">hoods</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-21-1.png" width="480"></p>
<p>We can cluster the cells by their neighborhood probability
distribution using <code>clustByHood</code>, it is based on the k-means
algorithm and here we set k to 10.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clustByHood.html">clustByHood</a></span><span class="op">(</span><span class="va">spe</span>, pm_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">hoods</span><span class="op">)</span>, k <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p>We can see what are the neighborhood distributions look like in each
cluster using <code>plotProbDist</code>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotProbDist.html">plotProbDist</a></span><span class="op">(</span><span class="va">spe</span>, pm_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">hoods</span><span class="op">)</span>, by_cluster <span class="op">=</span> <span class="cn">TRUE</span>, plot_all <span class="op">=</span> <span class="cn">TRUE</span>, show_clusters <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-23-1.png" width="480"></p>
<p>We can plot the clusters on the tissue slide, agian using
<code>plotTissue</code>.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotTissue.html">plotTissue</a></span><span class="op">(</span><span class="va">spe</span>, color <span class="op">=</span> <span class="va">clusters</span><span class="op">)</span></span></code></pre></div>
<p><img src="Quick_start_files/figure-html/unnamed-chunk-24-1.png" width="560"></p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] scico_1.5.0                 SpatialExperiment_1.10.0   </span></span>
<span><span class="co">##  [3] SingleCellExperiment_1.22.0 SummarizedExperiment_1.30.2</span></span>
<span><span class="co">##  [5] Biobase_2.60.0              GenomicRanges_1.52.0       </span></span>
<span><span class="co">##  [7] GenomeInfoDb_1.36.3         IRanges_2.34.1             </span></span>
<span><span class="co">##  [9] S4Vectors_0.38.1            BiocGenerics_0.46.0        </span></span>
<span><span class="co">## [11] MatrixGenerics_1.12.3       matrixStats_1.0.0          </span></span>
<span><span class="co">## [13] hoodscanR_0.99.1            ggplot2_3.4.3              </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] bitops_1.0-7              rlang_1.1.1              </span></span>
<span><span class="co">##  [3] magrittr_2.0.3            clue_0.3-64              </span></span>
<span><span class="co">##  [5] GetoptLong_1.0.5          compiler_4.3.1           </span></span>
<span><span class="co">##  [7] DelayedMatrixStats_1.22.6 png_0.1-8                </span></span>
<span><span class="co">##  [9] systemfonts_1.0.4         vctrs_0.6.3              </span></span>
<span><span class="co">## [11] stringr_1.5.0             shape_1.4.6              </span></span>
<span><span class="co">## [13] pkgconfig_2.0.3           crayon_1.5.2             </span></span>
<span><span class="co">## [15] fastmap_1.1.1             magick_2.7.5             </span></span>
<span><span class="co">## [17] XVector_0.40.0            scuttle_1.10.2           </span></span>
<span><span class="co">## [19] labeling_0.4.3            utf8_1.2.3               </span></span>
<span><span class="co">## [21] rmarkdown_2.24            ragg_1.2.5               </span></span>
<span><span class="co">## [23] purrr_1.0.2               xfun_0.40                </span></span>
<span><span class="co">## [25] zlibbioc_1.46.0           cachem_1.0.8             </span></span>
<span><span class="co">## [27] beachmat_2.16.0           jsonlite_1.8.7           </span></span>
<span><span class="co">## [29] rhdf5filters_1.12.1       DelayedArray_0.26.7      </span></span>
<span><span class="co">## [31] Rhdf5lib_1.22.1           BiocParallel_1.34.2      </span></span>
<span><span class="co">## [33] cluster_2.1.4             parallel_4.3.1           </span></span>
<span><span class="co">## [35] R6_2.5.1                  bslib_0.5.1              </span></span>
<span><span class="co">## [37] stringi_1.7.12            RColorBrewer_1.1-3       </span></span>
<span><span class="co">## [39] limma_3.56.2              jquerylib_0.1.4          </span></span>
<span><span class="co">## [41] Rcpp_1.0.11               iterators_1.0.14         </span></span>
<span><span class="co">## [43] knitr_1.44                R.utils_2.12.2           </span></span>
<span><span class="co">## [45] Matrix_1.5-4.1            tidyselect_1.2.0         </span></span>
<span><span class="co">## [47] abind_1.4-5               yaml_2.3.7               </span></span>
<span><span class="co">## [49] doParallel_1.0.17         codetools_0.2-19         </span></span>
<span><span class="co">## [51] lattice_0.21-8            tibble_3.2.1             </span></span>
<span><span class="co">## [53] withr_2.5.0               evaluate_0.21            </span></span>
<span><span class="co">## [55] desc_1.4.2                circlize_0.4.15          </span></span>
<span><span class="co">## [57] pillar_1.9.0              foreach_1.5.2            </span></span>
<span><span class="co">## [59] generics_0.1.3            rprojroot_2.0.3          </span></span>
<span><span class="co">## [61] RCurl_1.98-1.12           sparseMatrixStats_1.12.2 </span></span>
<span><span class="co">## [63] munsell_0.5.0             scales_1.2.1             </span></span>
<span><span class="co">## [65] glue_1.6.2                tools_4.3.1              </span></span>
<span><span class="co">## [67] locfit_1.5-9.8            RANN_2.6.1               </span></span>
<span><span class="co">## [69] fs_1.6.3                  rhdf5_2.44.0             </span></span>
<span><span class="co">## [71] grid_4.3.1                DropletUtils_1.20.0      </span></span>
<span><span class="co">## [73] edgeR_3.42.4              colorspace_2.1-0         </span></span>
<span><span class="co">## [75] GenomeInfoDbData_1.2.10   HDF5Array_1.28.1         </span></span>
<span><span class="co">## [77] cli_3.6.1                 textshaping_0.3.6        </span></span>
<span><span class="co">## [79] fansi_1.0.4               S4Arrays_1.0.6           </span></span>
<span><span class="co">## [81] ComplexHeatmap_2.16.0     dplyr_1.1.3              </span></span>
<span><span class="co">## [83] gtable_0.3.4              R.methodsS3_1.8.2        </span></span>
<span><span class="co">## [85] sass_0.4.7                digest_0.6.33            </span></span>
<span><span class="co">## [87] dqrng_0.3.1               rjson_0.2.21             </span></span>
<span><span class="co">## [89] farver_2.1.1              memoise_2.0.1            </span></span>
<span><span class="co">## [91] htmltools_0.5.6           pkgdown_2.0.7            </span></span>
<span><span class="co">## [93] R.oo_1.25.0               lifecycle_1.0.3          </span></span>
<span><span class="co">## [95] GlobalOptions_0.1.2</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Ning Liu, Jarryd Martin.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
